### PR TITLE
Adds kidneys

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Humans now have kidneys. Occasionally nature calls, just go to the nearest bathroom and do your stuff.
add: If you resent the time pressure that going to the bathroom once every 30 minutes has on your time, there are some alternatives. Become a species that doesn't have a renal system; have  your kidneys removed entirely (and just eat charcoal lest the toxins build up and kill you); get some shiny ELECTRONIC KIDNEYS from Robotics that are even more efficient than natural kidneys (at the cost of having mild side effects when EMP'd); go visit Chemistry and ask for some "Spaceperson's Friend".
/:cl:
- All human subspecies that require nutrition also have a renal system, with the exception of golems (because magical stone) and zombies (because their brain2energy conversion system is very efficient).
- Loss of the kidneys will cause a "waste products" reagent to build up, which is mildly toxic at high levels. It's possible to replace lost kidneys with a constant supply of charcoal.
- The timings are semi-randomised, but on average, a spessmen will get a notification at around 15 minutes, get an urgent notification at around 30 minutes, and start suffering from kidney failure at 45 minutes if ignored. Round start kidney filtration levels is also randomised so everyone isn't in sync with each other.
- I realise this is a controversial topic, so I'm gonna talk about some benefits.
### Benefits
- Non-traitorous uses for bathrooms; because people actually need to use them occasionally; this increases our PARANOIA levels
- Means that all this map space was actually used for something
- Silicons get to mock carbons
- Makes random isolated cult bases harder to maintain without occasionally popping over to the main station for "logistical reasons"
- Greytiders will remove their kidneys in protest and then probably throw them at security
  **\- This doesn't introduce any new emotes, effects or any visible messages.** This is not reagent farting, there will be no poop throwing, and the failure case (again, after around 45 minutes) is kidney failure.
- Ops will ask if "everyone's been to the bathroom, because we're not turning back once we've left okay?"
- Despite there being no new emotes or effects or messages, kidney function is completely separated from food intake

I am expecting a shitstorm of negative feedback, but I've tried really hard to keep things classy while still introducing a minor dynamic to the game.

### CODE REVIEW

**Comment by darkpaladin109** [comment]
Finally a use for toilets.

**Comment by PKPenguin321** [comment]
colons when

also you should be able to pee outside the bathroom because aliens needing human toilets is kinda stupid

**Comment by Jalleo** [comment]
Theyshould have their own toilet on the abductor vessel really.

**Comment by KorPhaeron** [comment]
What do I do if I'm exploring space or on lavaland or something?

**Comment by Jalleo** [comment]
Mining has a toilet last i knew

**Comment by Jalleo** [comment]
Probably not since each time they go back to mining station from a storm that can be a part of the busywork for the miner.

**Comment by KorPhaeron** [comment]
I just meant busywork for people in general. Slightly lowering the quality of everyones rounds for the sake of making a useless feature important (just like the chef except nobody plays as a plumber so there is even less reason here to make toilets """important"""")

**Comment by Zilenan91** [comment]
POOOOOOOOOOO :+1:

**Comment by coiax** [comment]
How is it "lowering" the quality of the round exactly?

**Comment by coiax** [comment]
Am I distracting people from their rampant murderboners and ash lizard hunts?

**Comment by KorPhaeron** [comment]
Making me run to an arbitrary room once when I'm in the middle of doing something I care about vs not running to that room once

**Comment by coiax** [comment]
You have like a 30 minute window, there's no rush.

**Comment by Zilenan91** [comment]
So if I'm understanding it right this just makes you run to a room, press a button, and then you're good for the rest of the round? No poop reagents you can mess around with or anything?

**Comment by xxalpha** [comment]
new toilet meta soon

**Comment by Zilenan91** [comment]
Also if people are unable to poop anywhere but in toilets this will make it so traitors C4 all the toilets so that the shuttle has to be called because otherwise everyone will die. Not necessarily a bad thing since it's hilarious but you shouldn't need a toilet to pass your stool.

**Comment by PKPenguin321** [comment]
yes. it is an unfortunate truth that poo cant really be added without allowing one to poo on the floor

**Comment by coiax** [comment]
There are currently no "poop" reagents, as you put them. I avoided anything like that to try to keep it relatively clean. I looked at what the sims does, and tried to copy that the best I can. And then ignored part of it.

>Also if people are unable to poop anywhere but in toilets this will make it so traitors C4 all the toilets so that the shuttle has to be called because otherwise everyone will die.

this sounds like a really terrible yet amazing plot. Also remember the space hotel has bathrooms, along with a bunch of runs.

**Comment by RemieRichards** [comment]
There isn't poo but there is however, bright yellow (#EED202) "waste" reagents.
Take a wild guess what this is.

**Comment by Zilenan91** [comment]
Sims are different though. SS13 isn't a sim, it's something where people run around and interact with round antagonists, they're not really remotely similar, and requiring people to walk into a room, press a button, and walk out isn't really fun or interesting.

Heck, even Lifeweb did poop right. It's basically the same thing, you have to walk into a room and press a button a few times and poop comes out, but you can actually use the poop for stuff. You can mix it into food to poison people, coat your weapon with it to give them diseases, and throw it at people to make them all stinky and make them sad. You should be able to at least throw your poop or as a chef coat your food with it for traitory stuff.

**Comment by InsaneHyena** [comment]
We goon now?

**Comment by coiax** [comment]
Waste products are now purple, because they're ABSTRACT.

**Comment by MMMiracles** [comment]
poo in the loo

**Comment by InsaneHyena** [comment]
Literally designated shitting PR

**Comment by AnturK** [comment]
While topical, please keep shitposting to minimum.

Anyway i agree with Kor this is another tedious meter to track  without much gameplay influence. Also this is kinda silly to add without making the toilets constructable.

**Comment by KingDairy** [comment]
alright but can you extract it?
can you kill people by injecting them with purpl stuph and then say some liner about them being full of shit?

**Comment by AdamElTablawy** [comment]
virology disease that gives diarrhea as a symptom when?

**Comment by TechnoAlchemisto** [comment]
literally no point and does not make the game more fun in any way

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 1]
```diff
@@ -751,4 +750,31 @@
...
+	color = "#8E3975"
+
+/datum/reagent/toxin/waste_product/overdose_process(mob/living/M)
+	// When built up in excess, slowly converts to toxin
+	// but at a rate slower than its normal buildup in humans
+	var/toxlover = FALSE
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.dna && H.dna.species && (TOXINLOVER in H.dna.species.specflags))
+			toxlover = TRUE
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: Replace with
`toxlover = (H.dna && H.dna.species && (TOXINLOVER in H.dna.species.specflags))`

**Comment by PKPenguin321** [comment]
upthumb revoked. if you're gonna add pee pee poo poo do it all the way, don't half ass it like this. this is the least fun way to do it.

**Comment by Fox-McCloud** [comment]
Invariably someone is going to add peeing on the floor because of this.

**Comment by GunHog** [comment]
Ash Walkers will need NORENAL or a little hole so they can take care of things as well.

**Comment by AlexCrimson** [comment]
Sounds fine to me, honestly. Whilst it isnt a needed feature, i like the idea that people need to take a few seconds to find a toilet and leave themselves vulnerable. This doesnt sound too bad to me.

**Comment by WJohn** [comment]
We are officially becoming the ship. Washing your hands or taking a shower to keep up your hygiene will become mandatory.

And I'm okay with that.

**Comment by Zilenan91** [comment]
Eh, the only reason needs exist in The Ship are to give people who have lived for a very long time and are ahead vulnerabilities where they can be ambushed unlike people who have been killed and respawned.

**Comment by optimumtact** [comment]
I told you that waste would not be considered, I'm disappointed you chose not to listen

**Comment by coiax** [comment]
Overstepping your bounds again, none of the other maintainers saw fit to close

**Comment by optimumtact** [comment]
I am not overstepping my bounds as a maintainer I have full authority to close a PR when and where I choose.

**Comment by AlexCrimson** [comment]
Is there a reason you closed this? Like will it hurt the state of the server or something? I assume you have a reason that isnt "i dont like it". You may have the ability to shut PRs down, but i imagine you need to give a valid reason, just like when Admins require a reason to Rule 0 someone.

**Comment by Iamgoofball** [comment]
S E P A R A T I O N
O F
C O D E B A S E
A N D
S E R V E R

**Comment by optimumtact** [comment]
Waste will not be considered.

**Comment by AlexCrimson** [comment]
Thats not a reason but ok, if you are going to be like that...

**Comment by coiax** [comment]
Maintainers are allowed to shut a PR down for any and no reason, it's dem rules. But it's not the first time oranges has overreached, and it won't be the last.

**Comment by ike709** [comment]
dat feel when oranges closes something presumably because he doesn't like it

seriously dude give an _actual_ reason at least

**Comment by KorPhaeron** [comment]
A maintainer not liking it is an actual reason to close it though, games are very subjective

**Comment by ike709** [comment]
That's pretty shit then (Pun not intended). Unless something is universally disliked, maintainers should at least elaborate as to why they dislike and close a PR instead of just saying "I don't like this."

**Comment by optimumtact** [comment]
I said why

Human waste will not be considered for inclusion in the game.

What isn't clear about that? Just because we don't list all our rules doesn't mean we don’t' have them.

**Comment by Fox-McCloud** [comment]
>What isn't clear about that? Just because we don't list all our rules doesn't mean we don’t' have them.

This; every codebase pretty much has this, so it's *really* not shocking to see this on TG as well--there's quite a few unspoken rules of "never going to happen" PRs that if they get submitted to a codebase...just aren't going to happen, no matter what....anddddd that's going to vary from codebase to codebase.

**Comment by Iamgoofball** [comment]
posting in a legendary thread

**Comment by TechnoAlchemisto** [comment]
i think this is the thread that killed coiax

**Comment by optimumtact** [comment]
It was the subsequent irc ban for backtalking the maintainers.

**Comment by coiax** [comment]
Eh, the IRC ban was just the straw that broke the camel's back.

There was a sense that everyone was constantly unhappy with everything that I did; MSO shouting and swearing at me about shuttles, being de-texted for answering ahelps (and probably rattling the admin cage more than was healthy), almost certainly a bunch of things that I have just forgotten about now, but it was a minutia of endless pokes and prods and insults; death by a thousand cuts.

I had put my damned hardest into spessmen, but the complaints were never ending, and the people that seemed to appreciate me being around were few and far between. No doubt I was difficult, stubborn, insulting and arrogant. But I'm sure I was some nice things too, but modesty forbids me from listing them.

In the end working /tgstation/ was a way of me attempting to find some routine and tasks to deal with/ignore my severe depression, along with letting me enjoy my innate skills at coding. It was a fun period of my life, and I'm proud that I can still point to parts of the codebase that are definitely mine. I'm definitely a little sad that I left, but I'm not sure if it was healthy for me in the end. I definitely miss some players from the server, they know who they are, but playing the game without coding for it, would be... pretty silly.

**Comment by optimumtact** [comment]
it's not an easy job that's for sure.

**Comment by Iamgoofball** [comment]
honestly coiax, i'd say continue taking a break and come back when you feel ready

i did the same thing for awhile and it helped me keep going through shit like goofchem and goofR&D and goofcargo and goofholyfuckhowmuchhaveioverhauledinthisshittygamefuckmeineedalife

your code was good with the only exception being this PR

**Comment by Jordie0608** [comment]
Your time was short but sweet, just like your line lengths.

**Comment by PKPenguin321** [comment]
But not like your changelogs :^)

No but seriously coiax you may have done a lot of controversial stuff like endless drone buffs but at the end of the day you are a really good coder which I respect a lot. I hope you come back some day when you feel up to it

**Comment by RemieRichards** [comment]
@coiax for what it's worth (given I dropped the hammer on you), I didn't want you gone permanently.

**Comment by EgorDinamit** [comment]
Legendary PR.

**Comment by optimumtact** [comment]
image

**Comment by Kurgis** [comment]
Kidneys? _Oh, cool, I wonder what kind of chemical and metabolism effects OP made for_ - okay it's just piss.

**Comment by Iamgoofball** [comment]
10 year anniversary coming up

**Pull request closed without merging**